lib/web_ui/lib/src/engine/history/history.dart

```diff
@@ -222,45 +214,70 @@ class MultiEntriesBrowserHistory extends BrowserHistory {
 ///  * [MultiEntriesBrowserHistory], which is used when the framework uses a
 ///    Router for routing.
 class SingleEntryBrowserHistory extends BrowserHistory {
+  SingleEntryBrowserHistory({required this.urlStrategy}) {
+    final UrlStrategy? strategy = urlStrategy;
+    if (strategy == null) {
+      return;
+    }
+
+    _setupStrategy(strategy);
+
+    final String path = currentPath;
+    if (_isFlutterEntry(html.window.history.state)) {
+      // This could happen if the user, for example, refreshes the page. They
+      // will land directly on the "flutter" entry, so there's no need to setup
+      // the "origin" and "flutter" entries, we can safely assume they are
+      // already setup.
```

Comment:
I'd do `if (! _isFlutterEntry`, remove the `else`, and rewrite the comment from the opposite point of view, e.g. "An entry may not have come from Flutter, for example, when...".